src/utils.ts

```diff
@@ -1,7 +1,7 @@
 export function undefinedIfEmpty<T>(obj: T): T | undefined {
   if (typeof(obj) === 'string' && obj === '') { return undefined; }
   if (Array.isArray(obj) && obj.length === 0) { return undefined; }
-  if (typeof(obj) === 'object' && Object.keys(obj).length === 0) { return undefined; }
+  if (typeof(obj) === 'object' && Object.keys(obj).length === 0 || Object.values(obj).filter(x => x).length === 0) { return undefined; }
```

Comment:
Can we add parentheses to make the intended order of ops here clearer? A unit test would also help here

Reply:
Added parentheses. As for unit test, this is already covered by our snapshot tests.